### PR TITLE
Ipelet for k-nearest neighbor graph

### PR DESCRIPTION
**Pull request by dfunke**
## Summary of Changes

This PR adds an Ipelet to draw the nearest-neighbor graph. It is based on the EMST Ipelet.

## Release Management

* Affected package(s): CGAL_Ipelets
* Issue(s) solved (if any): 
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature):
* License and copyright ownership: as you require

### CODE REVIEW

**Comment by mglisse** [comment]
Ah, I thought this would be a kNN graph with an input parameter k, but this is just the case k=1. It looks like there is an overload of `CGAL::nearest_neighbors` that takes a parameter `k`, if you decide to generalize the ipelet.

File: CGAL_ipelets/demo/CGAL_ipelets/nng.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  Triangulation t(pt_list.begin(), pt_list.end());
+
+  for(auto v = t.finite_vertices_begin();
+       v != t.finite_vertices_end();
+       ++v){
+
+    auto nn = CGAL::nearest_neighbor(t, v);
+    draw_in_ipe(Kernel::Segment_2(v->point(), nn->point()));
```
Comment (mglisse) [inline CGAL_ipelets/demo/CGAL_ipelets/nng.cpp:55]: It looks like if A is the nearest neighbor of B and vice versa, we can add the segment twice. Is that a feature (oriented graph), a "who cares", or a bug?

File: CGAL_ipelets/demo/CGAL_ipelets/nng.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  std::list<Point_2> pt_list;
+
+  read_active_objects(
+    CGAL::dispatch_or_drop_output<Point_2>(
+      std::back_inserter(pt_list)
+    )
+  );
+
+  if (pt_list.empty()) {
+    print_error_message("No mark selected");
```
Comment (mglisse) [inline CGAL_ipelets/demo/CGAL_ipelets/nng.cpp:44]: What happens if `pt_list` has a single point?

Comment (dfunke) [reply CGAL_ipelets/demo/CGAL_ipelets/nng.cpp:46]: With the move to the kNNG the return vector is simply empty

File: CGAL_ipelets/demo/CGAL_ipelets/nng.cpp (hunk 1 above)
Comment (dfunke) [reply CGAL_ipelets/demo/CGAL_ipelets/nng.cpp]: This is a feature: if the user adds forward arrows to the output, then it the edges correctly point at A and B

**Comment by dfunke** [comment]
Thank you for the excellent suggestion. I generalized the Ipelet to compute the kNNG.

> Ah, I thought this would be a kNN graph with an input parameter k, but this is just the case k=1. It looks like there is an overload of `CGAL::nearest_neighbors` that takes a parameter `k`, if you decide to generalize the ipelet.

**Comment by afabri** [comment]
Can you please change the name to something with "nearest_neighbors" in it.

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-116

**Pull request merged**